extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/cors/CORSFilter.java

```diff
@@ -21,6 +25,24 @@
 
 public class CORSFilter implements Handler<RoutingContext> {
 
+    static final Set<String> SIMPLE_HEADERS;
```

Comment:
@stuartwdouglas Why does this CORS filter need to worry about these subset of well known headers and content types ? Not all the headers/media types are listed here, what is the plan ? They are not used anywhere in the filter code just yet and I'm not sure CORS filter should be doing the related checks for the simple requests...

Reply:
Right, do you mean enforce that only those headers which are configured to be allowed, do make it during the simple request ? It can be hard to do though, as some headers like User-Agent, content negotiation ones, transfer encoding, etc can be added. Not sure we need to deal with it... For the preflight we have a list of allowed headers if needed

Reply:
I need to delete this. These headers are called out in the fetch spec as being explicitly CORS safe, and I originally thought I would block requests based on allowed headers, but that is not actually how Access-Control-Allowed-Headers works. 

This specifies the headers that the JavaScript is allowed to set on the request, not the allowed headers for the request, as the browser will set significatly more headers than what is specified in the allowed headers.

Reply:
Sure, makes sense, I can delete it for now and we can revisit